### PR TITLE
feat(alerts): Use metrics as crash free session dataset

### PR DESCRIPTION
**Pull request by scttcper**
Adds the feature flag `organizations:alert-crash-free-metrics`

### CODE REVIEW

File: static/app/views/alerts/incidentRules/ruleForm/index.tsx [hunk 1]
```diff
@@ -686,7 +687,7 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
         header={
           <ChartHeader>
             <AlertName>{AlertWizardAlertNames[alertType]}</AlertName>
-            {dataset !== Dataset.SESSIONS && (
```
Comment (scttcper) [inline static/app/views/alerts/incidentRules/ruleForm/index.tsx:689]: i think this is wrong

File: static/app/views/alerts/wizard/index.tsx [hunk 2]
```diff
@@ -74,10 +75,18 @@ class AlertWizard extends Component<Props, State> {
...
+    let metricRuleTemplate: WizardRuleTemplate | undefined =
+      AlertWizardRuleTemplates[alertOption];
     const isMetricAlert = !!metricRuleTemplate;
     const isTransactionDataset = metricRuleTemplate?.dataset === Dataset.TRANSACTIONS;
 
+    if (
+      organization.features.includes('alert-crash-free-metrics') &&
+      metricRuleTemplate?.dataset === Dataset.SESSIONS
+    ) {
+      metricRuleTemplate = {...metricRuleTemplate, dataset: Dataset.METRICS};
```
Comment (taylangocmen) [inline static/app/views/alerts/wizard/index.tsx:87]: Nit: can be `metricRuleTemplate.dataset = Dataset.METRICS;`

**Review by taylangocmen — APPROVED** [review]
lgtm

File: static/app/views/alerts/wizard/index.tsx (hunk 2 above)
Comment (scttcper) [reply static/app/views/alerts/wizard/index.tsx:87]: i think i want to avoid modifying `AlertWizardRuleTemplates`

**Pull request merged**